examples/pennylane_run_quantum_natural_gradient.py

```diff
@@ -0,0 +1,356 @@
+r"""
+
+.. _quantum_natural_gradient:
+
+Quantum natural gradient
+========================
+
+This example demonstrates the quantum natural gradient optimization technique
+for variational quantum circuits, originally proposed in
+`Stokes et al. (2019) <https://arxiv.org/abs/1909.02108>`__.
+
+Background
+----------
+
+The most successful class of quantum algorithms for use on near-term noisy quantum hardware
+is the so-called variational quantum algorithm. In variational quantum algorithms,
+a low-depth parametrized quantum circuit ansatz is chosen, and a problem-specific
+Hermitian observable measured. A classical optimization loop is then used to find
+the set of quantum parameters that *minimize* a particular measurement statistic
+of the quantum device. Examples of such algorithms include the :ref:`variational quantum
+eigensolver (VQE) <vqe>`, the `quantum approximant optimization algorithm (QAOA) <https://arxiv.org/abs/1411.4028>`__,
+and :ref:`quantum neural networks (QNN) <quantum_neural_net>`.
+
+Due to the inherent noise of near-term quantum hardware, most recent implementations
+of variational quantum algorithms have used gradient-free classical optimization
+methods, such as Nelder-Mead. However, the parameter-shift rule for
+analytic quantum gradients (as implemented in PennyLane) has allowed for
+stochastic gradient descent of variational quantum algorithms on quantum
+hardware. Once caveat though that has been surfaced with gradient descent
+is the issue of learning rate or step-size --- how do we choose the optimal
+step size for our variational quantum algorithms, to ensure successful and
+efficient optimization?
+
+The natural gradient
+^^^^^^^^^^^^^^^^^^^^
+
+In standard gradient descent, each optimization step is given by
+
+.. math:: \theta_{t+1} = \theta_t -\eta \nabla \mathcal{L}(\theta),
+
+where :math:`\mathcal{L}(\theta)` is the loss as a function of
+the parameters :math:`\theta`, and :math:`\eta` is the learning-rate
+or step size. In essence, each optimization step calculates a vector of
+steepest descent direction around the local value of :math:`\theta_t`
+in the parameter space, and updates :math:`\theta_t\rightarrow \theta_{t+1}`
+in by this vector in euclidean space.
+
+The problem with the above approach is that each optimization step
+is highly dependent on the euclidean geometry of the parameter space.
+The parametrization chosen is not necessarily unique; the scale of the
+parameters, or even on the parametrization used, could easily be modified.
+If we instead consider the optimization problem as a distribution of possible
+output values given an input, a better approach is to instead perform the
+gradient descent in the *distribution space*, which is invariant with respect

```

Comment:
Which distribution are you talking about? Explain what the random variable is here, or how this relates to quantum states.

Reply:
I suppose I am trying to describe this: https://en.wikipedia.org/wiki/Likelihood_function :thinking:

Reply:
Maybe just rigorously say what the distribution/likelihood refers to? I don't think you need to explain Adam and Eve (could refer to wikipedia or paper), but just so that all information is in principle in the text so that a maths-minded person could piece things together?

Reply:
I've added in a link as you suggest!